app/src/main/java/org/oppia/app/parser/StringToRatioParser.kt

```diff
@@ -25,15 +26,19 @@ class StringToRatioParser {
    * using [getRealTimeAnswerError], instead.
    */
   fun getSubmitTimeError(text: String, numberOfTerms: Int): RatioParsingError {
-    if (!text.matches(invalidRatioRegex))
+    val normalized = text.normalizeWhitespace()
+    // Check for invalid format before parsing otherwise it would throw an error instead.
+    if (!normalized.matches(invalidRatioRegex)) {
       return RatioParsingError.INVALID_FORMAT
-    val ratio = parseRatioOrThrow(text)
-    return if (numberOfTerms != 0 && ratio.ratioComponentCount != numberOfTerms) {
-      RatioParsingError.INVALID_SIZE
-    } else if (ratio.ratioComponentList.contains(0)) {
-      RatioParsingError.INCLUDES_ZERO
-    } else {
-      RatioParsingError.VALID
+    }
+    val ratio = parseRatioOrThrow(normalized)
+    return when {
+      ratio == null -> RatioParsingError.INVALID_FORMAT
```

Comment:
done

Reply:
Actually it would throw out the error while parsingRatio 
For e.g. if the input is 1:2:3: and we submit that answer in that case first it will parse the error then it would match the regex pattern but it should be first matching the regex pattern then it should be parsing the ratio. Does that make sense ?

Reply:
This has been updated after our chat. Thanks for the clarification